generic_config_updater/patch_sorter.py

```diff
@@ -399,10 +399,10 @@ def _get_paths_recursive(self, config, pattern_tokens, matching_tokens, idx, com
         if token == "*":
             matching_keys = config.keys()
         elif token.startswith("*|"):
-            suffix = token[2:]
+            suffix = token[1:] # the suffix will be `|...`
             matching_keys = [key for key in config.keys() if key.endswith(suffix)]
         elif token.endswith("|*"):
-            prefix = token[:-2]
+            prefix = token[:-1] # the prefix will be `...|`
```

Comment:
Without this change the pattern `*|10` would match `Ethernet1|10` and `Ethernet1|11110`